src/web/app/app.tsx

```diff
@@ -0,0 +1,265 @@
+
+import * as React from 'react';
+import { useSearchParams } from 'react-router-dom';
+
+import { type StudyModel, type StudyListModel } from './study_model';
+import { type FeatureModel, type ExperimentModel } from './experiment_model';
+import { type StudyFilter } from '../../core/study_processor';
+import { SeedType } from '../../core/base_types';
+import { loadSeedDataAsync } from './seed_loader';
+import { SearchParamManager } from './search_param_manager';
+
+export function FeatureItem(props: {
+  feature: FeatureModel;
+  className: string;
+}) {
+  return (
+    <a
+      className={props.className}
+      target="blank"
+      rel="noreferrer"
+      href={props.feature.link}
+    >
+      {props.feature.name}
+    </a>
+  );
+}
+
+export function FeatureList(props: {
+  title: string;
+  className: string;
+  features: FeatureModel[];
+}) {
+  if (props.features.length === 0) {
+    return <></>;
+  }
+  const features = props.features.map((f) => (
+    <li key={f.name}>
+      <FeatureItem feature={f} className={props.className} />
+    </li>
+  ));
+  return (
+    <ul className="study-meta">
+      <span>{props.title}: </span>
+      {features}
+    </ul>
+  );
+}
+
+export function ExperimentItem(props: { exp: ExperimentModel }) {
+  const paramsList = props.exp.parameters().map((p) => <li key={p}>{p}</li>);
+  const classes =
+    'list-group-item ' +
+    (props.exp.isMajorGroup() ? 'major-exp-item' : 'exp-item');
+  return (
+    <li className={classes}>
+      {props.exp.name()} ({props.exp.weight()}%)
+      <FeatureList
+        title="Enabled features"
+        className="enabled-feature"
+        features={props.exp.enabledFeatures()}
+      />
+      <FeatureList
+        title="Disabled features"
+        className="disabled-feature"
+        features={props.exp.disabledFeatures()}
+      />
+      {paramsList.length > 0 && (
+        <ul className="study-meta">
+          <span>Parameters: </span>
+          {paramsList}
+        </ul>
+      )}
+    </li>
+  );
+}
+
+export function PropertyList(props: {
+  caption: string;
+  list: string[] | null | undefined;
+}) {
+  if (props.list == null || props.list.length === 0) return <></>;
+  return (
+    <ul className="study-meta">
+      <span>{props.caption}: </span>
+      {props.list.map((c) => (
+        <li key={c}>{c}</li>
+      ))}
+    </ul>
+  );
+}
+
+export function IncludeExcludeList(props: {
+  caption: string;
+  include: string[] | null | undefined;
+  exclude: string[] | null | undefined;
+}) {
+  return (
+    <>
+      <PropertyList caption={props.caption} list={props.include} />
+      <PropertyList caption={'Exclude' + props.caption} list={props.exclude} />
+    </>
+  );
+}
+
+export function StudyItem(props: { study: StudyModel; filter: StudyFilter }) {
+  const filter = props.study.filter();
+  const minVersion = props.study.filter()?.min_version ?? '';
+  const maxVersion = props.study.filter()?.max_version ?? '';
```

Comment:
These lines are meant to use `filter` variable, not call `props.study.filter()` again?

Reply:
done